### PR TITLE
Add testing for save/restore snapshots with geo columns

### PR DESCRIPTION
**Pull request by wolffcm**
There is a companion pro pull request:

https://github.com/VoltDB/pro/pull/1044

### CODE REVIEW

**Comment by BillWhite** [comment]
This all seems plausible as well.

File: src/frontend/org/voltdb/VoltTable.java [hunk 1]
```diff
@@ -921,6 +924,17 @@ private void addColumnValue(Object value, VoltType columnType, int col) {
...
                 }
 
+                case POINT: {
+                    if (value instanceof PointType) {
+                        PointType pt = (PointType)value;
+                        pt.flattenToBuffer(m_buffer);
+                    }
+                    else {
+                        throw new ClassCastException();
+                    }
```
Comment (paulmartel) [inline src/frontend/org/voltdb/VoltTable.java]: This is a strangely verbose way of getting a ClassCastException.
You get the same behavior if you take out the "if (value instanceof PointType) {".
Is there a rationale for being purposely more verbose?
It looks like this was cribbed from a different case that wanted to try multiple object formats before failing with the ClassCastException -- it HAD a rationale THERE -- wanting to have parallel code paths for the different input types and not wanting to have to catch the first ClassCastException for the chance to retry.

File: src/frontend/org/voltdb/VoltType.java [hunk 2]
```diff
@@ -742,7 +742,8 @@ public static boolean isNullVoltType(Object obj)
         if ((obj == null) ||
             (obj == VoltType.NULL_TIMESTAMP) ||
             (obj == VoltType.NULL_STRING_OR_VARBINARY) ||
-            (obj == VoltType.NULL_DECIMAL))
+            (obj == VoltType.NULL_DECIMAL) ||
+            (obj == VoltType.NULL_POINT))
```
Comment (paulmartel) [inline src/frontend/org/voltdb/VoltType.java]: Added handling for NULL_POINT, but what about GEOGRAPHY?

**Comment by paulmartel** [comment]
Either I missed something or the only thing being tested here is save/restore for NULL POINT/GEOGRAPHY values. SO, seems like it could use more coverage.

**Comment by paulmartel** [comment]
Now I see the pro coverage for the pro feature. DOH! nevermind. This looks good.

File: src/frontend/org/voltdb/VoltTable.java (hunk 1 above)
Comment (wolffcm) [inline src/frontend/org/voltdb/VoltTable.java]: You are right, of course.  I simplified the code like you suggested.

**Comment by wolffcm** [comment]
Rebased branch...

**Pull request closed without merging**